cmd/skopeo/sync.go

```diff
@@ -186,6 +198,16 @@ func destinationReference(destination string, transport string) (types.ImageRefe
 			return nil, errors.Wrapf(err, "Error creating directory for image %s", destination)
 		}
 		imageTransport = directory.Transport
+	case oci.Transport.Name():
+		dir := path.Dir(destination)
+		if err := os.MkdirAll(destination, 0755); err != nil {
+			return nil, errors.Wrapf(err, "Error creating directory:%s for image %s", dir, destination)
+		}
+		if ref, err := reference.ParseNormalizedNamed(destination); err != nil {
+			tagref, _ := reference.TagNameOnly(ref).(reference.Tagged)
+			destination = fmt.Sprintf("%s:%s", reference.Path(ref), tagref)
```

Comment:
This is pre-existing, but now is a good time to clean up:

Let’s stop formatting `imageTransport` and `destination` strings only to parse them below in `imageTransport.ParseReference`; instead, let’s have every case create `destRef` directly (e.g. in this case using `oci.NewReference`).